libs/storage/src/ipfs/ipfs.service.spec.ts

```diff
@@ -16,18 +16,175 @@ const mockHttpCommonConfigProvider = GenerateMockConfigProvider<IHttpCommonConfi
   httpResponseTimeoutMS: 3000,
 });
 
+async function* bufferGenerator(values: number[]): AsyncIterable<Uint8Array> {
+  yield new Uint8Array(values);
+}
+
+async function* pinLsGenerator(pins: PinLsResult[]): AsyncIterable<PinLsResult> {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const pin of pins) {
+    yield pin;
+  }
+}
+
 describe('IpfsService Tests', () => {
   let service: IpfsService;
+  let ipfs: KuboRPCClient;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [IpfsService, mockIpfsConfigProvider, mockHttpCommonConfigProvider],
     }).compile();
 
     service = module.get<IpfsService>(IpfsService);
+    ipfs = (service as unknown as any).ipfs;
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  describe('getPinned', () => {
+    it('should return empty buffer if resource is not pinned', async () => {
+      jest.spyOn(service, 'isPinned').mockResolvedValueOnce(false);
```

Comment:
why the different response between getPinned (empty buffer) and getInfo (throw error)?

Reply:
That's the way `getPinned` was initially; I didn't bother changing it; it's used in `content-watcher` as well and I didn't want to increase the scope of this PR to `content-watcher`.

I modified `getInfo` to throw an error because it's only called in one spot, and that spot just checked for the error response and wrapped it in a thrown error anyway, so I cut out the middle man 😄